primitives/src/merkle_tree/macros.rs

```diff
@@ -105,6 +99,14 @@ macro_rules! impl_merkle_tree_scheme {
                 }
                 proof.borrow().verify_membership_proof::<H>(root.borrow())
             }
+
+            fn iter(&self) -> MerkleTreeIter<Self::Element, Self::Index, Self::NodeValue> {
+                MerkleTreeIter::new(&self.root)
+            }
+
+            fn into_iter(self) -> MerkleTreeIntoIter<Self::Element, Self::Index, Self::NodeValue> {
+                MerkleTreeIntoIter::new(self.root)
+            }
```

Comment:
Slightly more idiomatic would be to implement `IntoIter<Item = (&I, &E)> for &Self` and `IntoIter<Item = (I, E)> for Self`, similar to how `Vec` does it

Reply:
Good idea, but will keep `iter()` because it's not supported by any traits. [937c573](https://github.com/EspressoSystems/jellyfish/pull/475/commits/937c57379424b074bfe5caf61d3b7538a8b8deb8)